test/src/e2e_vm_tests/mod.rs

```diff
@@ -936,22 +934,78 @@ fn parse_test_toml(path: &Path, run_config: &RunConfig) -> Result<TestDescriptio
     let mut file_check = FileCheck(toml_content_str.clone());
     let checker = file_check.build()?;
 
-    let toml_content = toml_content_str.parse::<toml::Value>()?;
+    let mut toml_content = toml_content_str.parse::<toml::Value>()?;
 
     if !toml_content.is_table() {
         bail!("Malformed test description.");
     }
 
+    // use test.toml as base if this test has a suffix
+    {
+        let toml_content_map = toml_content.as_table_mut().unwrap();
+        if path.file_name().and_then(|x| x.to_str()) != Some("test.toml") {
+            let base_test_toml = path.parent().unwrap().join("test.toml");
+            let base_test_toml = std::fs::read_to_string(base_test_toml)?;
+            let base_toml_content = base_test_toml.parse::<toml::Value>()?;
+            match base_toml_content {
+                toml::Value::Table(map) => {
+                    for (k, v) in map {
+                        let _ = toml_content_map.entry(&k).or_insert(v);
+                    }
+                }
+                _ => todo!(),
```

Comment:
Shouldn't this be unreachable if we sort the result of `glob::glob` by path? In that case, the base `test.toml` will always be parsed before any feature tomls and will fail at that point if it cannot be parsed. In that case we know that here it will always be available an we can simply `expect` and perhaps skip the `match` completely.